src/storage/btree.c

```diff
@@ -4280,13 +4302,171 @@ btree_read_record (THREAD_ENTRY * thread_p, BTID_INT * btid, PAGE_PTR pgptr, REC
+#define check_validate(bts) ;
+#endif
+
+#if defined(IMPROVE_RANGE_SCAN_IN_BTREE)
+static int
+btree_read_record_in_leafpage (THREAD_ENTRY * thread_p, PAGE_PTR pgptr, int copy_key, BTREE_SCAN * bts)
+{
+  int error, n_prefix;
+
+  assert (bts && pgptr && bts->key_record.type == REC_HOME);
+
+  bts->is_cur_key_compressed = false;
+  error =
+    btree_read_record_without_decompression (thread_p, &bts->btid_int, &bts->key_record, &(bts->cur_key),
+					     (void *) &bts->leaf_rec_info, BTREE_LEAF_NODE, &(bts->clear_cur_key),
+					     &bts->offset, copy_key);
+  if (error != NO_ERROR)
+    {
+      return error;
+    }
+
+  assert (btree_leaf_is_flaged (&bts->key_record, BTREE_LEAF_RECORD_FENCE) == false);
+  if (btree_leaf_is_flaged (&bts->key_record, BTREE_LEAF_RECORD_OVERFLOW_KEY)
+      || TP_DOMAIN_TYPE (bts->btid_int.key_type) != DB_TYPE_MIDXKEY)
+    {
+      return NO_ERROR;
+    }
+
+  assert (pgptr == bts->C_page && VPID_EQ (pgbuf_get_vpid_ptr (pgptr), &bts->C_vpid));
+
+  if (bts->common_prefix_size >= 0)
+    {
+      assert (VPID_EQ (&(bts->C_vpid), &bts->cur_common_prefix_page_vpid));
+#if defined(CHECK_VERIFY_COMMON_PREFIX_PAGE_INFO)
+      assert (LSA_EQ (&bts->cur_common_prefix_page_lsa, pgbuf_get_lsa (pgptr)));
+      assert (bts->common_prefix_size == btree_node_get_common_prefix (thread_p, &bts->btid_int, pgptr));
+#endif
+
+      if (bts->common_prefix_size > 0)
+	{
+	  bts->is_cur_key_compressed = true;
+	}
+      return NO_ERROR;
+    }
+
+  assert (bts->common_prefix_size == COMMON_PREFIX_UNKNOWN);
+
+  /* recalculate n_prefix */
+  n_prefix = btree_node_get_common_prefix (thread_p, &bts->btid_int, pgptr);
+
+  bts->common_prefix_size = n_prefix;
+#if defined(CHECK_VERIFY_COMMON_PREFIX_PAGE_INFO)
+  VPID_COPY (&(bts->cur_common_prefix_page_vpid), &(bts->C_vpid));
+  LSA_COPY (&bts->cur_common_prefix_page_lsa, pgbuf_get_lsa (pgptr));
+#endif
+  btree_clear_key_value (&(bts->clear_common_prefix_key), &(bts->common_prefix_key));
+
+  if (n_prefix > 0)
+    {
+      RECDES peek_rec;
+      LEAF_REC leaf_pnt;
+      int dummy_offset;
+
+      (void) spage_get_record (thread_p, pgptr, 1, &peek_rec, PEEK);
+      error =
+	btree_read_record_without_decompression (thread_p, &bts->btid_int, &peek_rec, &(bts->common_prefix_key),
+						 &leaf_pnt, BTREE_LEAF_NODE, &(bts->clear_common_prefix_key),
+						 &dummy_offset, PEEK_KEY_VALUE);
+      if (error != NO_ERROR)
+	{
+	  btree_clear_key_value (&(bts->clear_common_prefix_key), &(bts->common_prefix_key));
+	  return error;
+	}
+      assert (btree_leaf_is_flaged (&peek_rec, BTREE_LEAF_RECORD_FENCE));
+      assert (DB_VALUE_TYPE (&(bts->common_prefix_key)) == DB_TYPE_MIDXKEY);
+      bts->is_cur_key_compressed = true;
+    }
+  else
+    {
+      if (n_prefix < 0)
+	{
+	  assert_release (false);
 	  return n_prefix;
 	}
     }
 
   return NO_ERROR;
 }
 
+static void
+btree_make_complete_key_including_prefix (BTREE_SCAN * bts, DB_VALUE * common_prefix_key, int common_prefix_size)
```

Comment:
This function is no different from pr_midxkey_add_prefix and is not called much.
So I think it is unnecessary.
How about using pr_midxkey_add_prefix?

Reply:
Are you asking whether it can be implemented by calling pr_midxkey_add_prefix() directly in btree_check_decompress_key()?
It is possible if the question was asked in that sense.
However, there are plans to call this function directly in other issues in the future.

Reply:
The functions `btree_make_complete_key_including_prefix`, `btree_check_decompress_key`, and `pr_midxkey_add_prefix` have the same purpose.
However, there is no difference in the results of code execution.

The pr_midxkey_add_prefix function has different layers.
So my opinion was that the two functions btree_make_complete_key_including_prefix and btree_check_decompress_key should be reduced to one.

I understood that the purpose of the call was different.

The purpose is to express my opinion, so if you read it, please resolve it.